Makefile

```diff
@@ -31,3 +31,12 @@ docker-build:
 
 docker-push:
 	@docker push ${NAME}
+
+lint:
+	@echo "Running pre-commit hooks..."
+	@if command -v pre-commit >/dev/null 2>&1; then \
+		pre-commit run --all-files; \
+	else \
+		echo "pre-commit not found. Install with: pip install pre-commit"; \
+		exit 1; \
+	fi
```

Comment:
I do not want Python dependencies, can we convert whatever the pre-commit hook is supposed to be doing into a POSIX SH / Bash script? 

I can write it, just let me know what it should be doing if not obvious.

Reply:
yeah totally doable tbh. pre-commit basically just runs linters/formatters on staged files before commit. the main sauce is:

1. grab staged files w/ git diff --cached --name-only
2. filter by extension (.go, .md, whatever)
3. run your tools on those files as per the config file
4. exit 1 if any fail

the tricky bit pre-commit handles is the stash dance - it stashes unstaged changes so you're ONLY checking what's actually being committed: